src/value/mod.rs

```diff
@@ -878,64 +879,106 @@ mod generic_value_tests {
         }
     }
 
-    fn bool_case<E: Element>() -> Case<E> {
+    fn bool_case<E: Element>() -> Case<E>
+    where
+        E: Debug + PartialEq,
```

Comment:
Should we just add this super-trait on `Element` and friends?

Reply:
Yes. I will add it.